tests/ops/op_math/test_controlled_op.py

```diff
@@ -846,3 +872,99 @@ def pow(self, z):
         assert pow_op.control_wires == self.control_wires
         assert pow_op.control_values == self.control_values
         assert pow_op.work_wires == self.work_wires
+
+
+@pytest.mark.parametrize("diff_method", ["backprop", "parameter-shift", "finite-diff"])
+class TestDifferentiation:
```

Comment:
We could also test the differentiation of a controlled quantum function

Reply:
If I have time, I'll add some in.  Trying to just get the normal case working again now...